### DISCUSSION TITLE
Python 3.7+

### DISCUSSION THREAD

**Discussion by sincze** [bimmerconnected/bimmer_connected#558] [General] [answered]
I am running Bimmer Connected (CLI) on a Raspberry Pi 4.
That one has Python 3.7.3

I'm also using Domoticz that uses the Python framework. 
I think it should be possible to safely update 3.7 to python 3.8 without breaking Domoticz and Bimmer Connected.

To prevent accidents in my live setup I asked a friend to create a Docker that can run the Latest Bimmer Connected and push the **status** outcome into Mosquitto for further processing in Node-Red.

<img width="229" alt="image" src="https://github.com/bimmerconnected/bimmer_connected/assets/5776333/da093d73-4330-49dc-b403-fbfcad43cda7">

It looks promissing
<img width="744" alt="image" src="https://github.com/bimmerconnected/bimmer_connected/assets/5776333/fc2ac834-1238-4f84-8a30-1d4e915d882e">

If somebody else has a better idea, please let me know.

**Comment by rikroe** [comment]
Not sure what you are trying to do.

This library fully supports Python 3.8 to 3.11. Python 3.7 is not supported anymore.

To migrate, best uninstall the library using pip, install a newer version of python (preferably the latest available, as 3.8 will be out of support by soon as well).

**Reply by sincze** [reply]
What I try to do is not to break an existing Raspberry Pi (Domoticz) environment that still seems to depend on 3.7 :) 

Running BimmerConnected in Docker with Python 3.1x works just as well adjacent to my existing setup. It pushes the CLI json into MQTT so I can work with it in NodeRed.

**Reply by rikroe** [reply, marked as answer]
Judging from https://www.domoticz.com/wiki/Using_Python_plugins, a newer python version should work with demoticz.

As alternative (if you're only interested in the status) you could install it in a separate venv and just use the status CLI with the json flag.

**Reply by sincze** [reply]
Tnx @rikroe , Raspberry not that fancy with newer Python versions unfortunately (but possible indeed), for now it is working just fine. ;-) 

Other solution is a well known node-red version of bimmer-connected to control the thing. But 99% of the time for me it is just collecting the status of the machine. Tnx for your support.
